code/game/objects/effects/aliens.dm

```diff
@@ -200,35 +205,60 @@
 	var/node_range = NODERANGE
 	var/list/obj/effect/alien/weeds/connected_weeds
 
-/obj/effect/alien/weeds/node/Destroy()
-	for(var/obj/effect/alien/weeds/W in connected_weeds)
-		W.linked_node = null
-	..()
 
 /obj/effect/alien/weeds/node/New()
 	connected_weeds = new()
 	..(src.loc, src)
 
+/obj/effect/alien/weeds/node/Destroy()
+	for(var/obj/effect/alien/weeds/W in connected_weeds)
+		W.health -= 2 * get_dist(W, src)//so the further ones are more likely to disappear first
+		W.linked_node = null
+		spawn()
+			W.DieWeed()
+	..()
+
 /obj/effect/alien/weeds/New(pos, var/obj/effect/alien/weeds/node/N)
 	..()
 
 	if(istype(loc, /turf/space))
 		qdel(src)
 		return
 
+	for(var/obj/machinery/door/D in loc)
+		if(istype(D,/obj/machinery/door/mineral/resin))
+			continue
+		if(!D.density && !D.operating)
+			D.jammed = src
+			jammin = D
+
 	linked_node = N
 	if(linked_node)
 		linked_node.connected_weeds.Add(src)
 
 	if(icon_state == "weeds")icon_state = pick("weeds", "weeds1", "weeds2")
-	spawn(rand(150, 200))
+	spawn(rand(100, 250))
 		if(src)
 			Life()
 	return
 
-/obj/effect/alien/weeds/proc/Life()
-	if(timestopped) return 0 //under effects of time magick
+/obj/effect/alien/weeds/node/New()
+	..()
+	spawn(HEARTBEAT_RATE)
+		heartbeat()
+
+/obj/effect/alien/weeds/node/proc/heartbeat()
+	flick("weednode-heartbeat",src)
+
+	for(var/obj/effect/alien/weeds/W in connected_weeds)
+		if(W.health == 15)
+			spawn()
+				W.Life()
+	spawn(HEARTBEAT_RATE)
+		heartbeat()
```

Comment:
The point of heartbeat() is to allow the previously spawned weeds to keep expanding if a wall has been destroyed/a door has been openned. (also it has a flick animation so it looks like the node is alive every 30 seconds) What would if(!gcDestroyed) do here?